src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -135,7 +114,25 @@ public class PipelineCacheServer
 
                 if (!string.IsNullOrEmpty(cacheHitVariable))
                 {
-                    context.SetVariable(cacheHitVariable, result != null ? "true" : "false");
+                    if (result == null) {
+                        context.SetVariable(cacheHitVariable, "false");
+                    }  else  {
+                        context.Verbose($"Exact fingerprint: `{result.Fingerprint.ToString()}`");
+
+                        bool foundExact = false;
+                        foreach(var fingerprint in fingerprints)
+                        {
+                            context.Verbose($"This fingerprint: `{fingerprint.ToString()}`");
+
+                            if(fingerprint == result.Fingerprint)
+                            {
```

Comment:
so we're only going output inexact on wildcard related fingerprints and not exclusively fallback keys?

Reply:
A primary key can't be a wildcard because that's the name it saves as. Did that answer your question?